### ISSUE TITLE
[export] Cannot mutate tensors with frozen storage

### ISSUE THREAD

**Issue by zhxchen17** [pytorch/pytorch#127571]
### 🐛 Describe the bug

```	
    def test_inplace_masked_fill(self):
        class Model(torch.nn.Module):
            def __init__(self):
                super().__init__()
                self.dtype = torch.float32

            def forward(self, context_mask):
                mask = torch.full((tgt_len, tgt_len), torch.finfo(dtype).min, device=device)
                mask = mask.to(self.dtype)
                mask.masked_fill_(context_mask, torch.finfo(self.dtype).min)
                return mask

        model = Model()
        ep = export(model, (torch.ones(1, 1, 1, 1),))
```

Today this will fail with "Cannot mutate tensors with frozen storage" error because right now we ban all kinds of mutation on converted tensors from _to_copy(). 

Internal post: https://fb.workplace.com/groups/1075192433118967/permalink/1438891976749009/

Will have a discussion with @bdhirsh about the possible options moving forward.

### Versions

nightly

cc @avikchaudhuri @gmagogsfm @tugsbayasgalan @angelayi @suo @ydwu4

**Comment by arnav177** [comment]
The error "Cannot mutate tensors with frozen storage" occurs because recent changes prevent mutations on tensors converted with _to_copy(). To address this, we can clone the tensor before performing in-place operations to ensure it can be mutated.

ef test_inplace_masked_fill(self):
    class Model(torch.nn.Module):
        def _init_(self):
            super()._init_()
            self.dtype = torch.float32

        def forward(self, context_mask):
            mask = torch.full((tgt_len, tgt_len), torch.finfo(self.dtype).min, device=device)
            mask = mask.to(self.dtype)
            mask = mask.clone()  # Clone the tensor to ensure it can be mutated
            mask.masked_fill_(context_mask, torch.finfo(self.dtype).min)
            return mask

    model = Model()
    ep = export(model, (torch.ones(1, 1, 1, 1),))

**Comment by bhack** [comment]
I got the same error.. So is it expected?

**Comment by yushangdi** [comment]
Should be fixed now by https://github.com/pytorch/pytorch/pull/142420 when `self.dtype` is different from `mask.dtype`.
